### PR TITLE
153   implement the proper layout of the settings page

### LINKED ISSUE #153: Implement the proper layout of the settings page

### PR DESCRIPTION
**Pull request by Exclaminator**
Will close #153.

Do note that these options still have to be implemented.

### CODE REVIEW

File: resources/options.html [hunk 1]
```diff
@@ -43,120 +43,79 @@
...
+            <div class="card">
+                <div class="card-content">
+                    <span class="card-title">Mouse</span> <br>
+                    <p>
+                        While I'm reviewing pull requests on GitHub, the following actions involving my mouse may be tracked:
+                    </p>
+                    <ul id="mouseSubOptions">
+                        <li>
+                            <input type="checkbox" id="mouseHover" checked>
+                            <label for="mouseHover"> what elements I hover on.</label>
```
Comment (thervh70) [inline resources/options.html]: hover on or hover over?

File: resources/options.html [hunk 2]
```diff
@@ -43,120 +43,79 @@
...
+            <div class="card" id="data">
+                <div class="card-content">
+                    <span class="card-title">Data</span> <br>
+                    <p>
+                        While I'm reviewing pull requests on GitHub, the following actions may be done:
+                    </p>
+                    <ul id="dataSubOptions">
+                        <li>
+                            <input type="checkbox" id="dataTabs">
+                            <label for="dataTabs"> log when I swich active tabs. </label>
```
Comment (thervh70) [inline resources/options.html]: *switch

File: resources/options.html [hunk 3]
```diff
@@ -43,120 +43,79 @@
...
+                            <input type="checkbox" id="dataComments" checked>
+                            <label for="dataComments"> log the comments on a PR. </label>
+                        </li>
+                        <li>
+                            <input type="checkbox" id="dataKeystrokes" checked>
+                            <label for="dataKeystrokes"> log individual keystrokes / shortcuts on the PR page. </label>
+                        </li>
+                        <li>
+                            <input type="checkbox" id="dataHTML">
+                            <label for="dataHTML"> log the eintire pull request (the HTML body). </label>
```
Comment (thervh70) [inline resources/options.html]: *entire

**Comment by thervh70** [comment]
Besides the few spelling errors there are some more problematic things. Please examine the following 2 figures.
figure 1
figure 2
Some weird things are happening with the last 2 options. Please take a look into this.

File: resources/options.html [hunk 4]
```diff
@@ -43,120 +43,79 @@
...
+                            <input type="checkbox" id="mouseHover" checked>
+                            <label for="mouseHover"> what elements I hover over.</label>
+                        </li>
+                        <li>
+                            <input type="checkbox" id="mouseClick" checked>
+                            <label for="mouseClick"> what elements I click on. </label>
+                        </li>
+                        <li>
+                            <input type="checkbox" id="mouseScrolling" checked>
+                            <label for="mouseScrolling"> how I use the scroll bar. </label>
```
Comment (mpsijm) [inline resources/options.html]: People can also use the scroll wheel? :P Maybe make this a bit more general, like "when I scroll through the page"?

File: resources/options.js [hunk 5]
```diff
@@ -139,11 +139,13 @@ function addOptionClickEvents() {
 function dynamicalyHideOrShowSubelements(tohide) {
     if ($("#" + tohide[0]).prop('checked')) {
         for (var s in tohide[1]) {
-            $("#" + tohide[1][s]).prop('checked', true).removeAttr("disabled");
+            if ( $("#" + tohide[1][s]).attr("disabled")) {
```
Comment (mpsijm) [inline resources/options.js]: Please remove the space before `(` and `$`.

**Comment by mpsijm** [comment]
When reloading the options page, the disabled state is not restored.

https://gyazo.com/9ce65b97140cde7204341cf8ae857dc5

**Comment by Exclaminator** [comment]
@mpsijm are you certain that you have checked this on the last version?

**Comment by Exclaminator** [comment]
The last commit should fix this, good catch!

File: resources/options.html [hunk 6]
```diff
@@ -1,171 +1,117 @@
...
+        <meta name="analytics-browser-action" content="_self"/>
+    </head>
+    <body>
+        <nav>
+            <div class="nav-wrapper lime">
+                <div class="brand-logo center" style="height: auto; width: auto;">
+                    <a>Octopeer</a>
+                </div>
+                <ul class="right hide-on-small-only">
+                    <li><a href="../analytics/index.html">Analytics</a></li>
```
Comment (MathiasMeuleman) [inline resources/options.html]: Why remove the id here? Was purposely installed

File: resources/options.js [hunk 7]
```diff
@@ -59,37 +67,17 @@ function restoreOptionsAvailability() {
...
-}
-
-// Hides the option cards.
-function hideCards() {
-    for (var i = 0; i < cards.length; i++) {
-        cardElement(i).hide();
+// Restores the availability of the checkboxes, that are enabled / disabled by other checkboxes. using the logging value which is stored in chrome.storage.
+function restoreSubOptionsAvailability() {
+    for (var i = 0; i < onDisableHideAndDisable.length; i++) {
+        dynamicalyHideOrShowSubelements(onDisableHideAndDisable[i]);
```
Comment (MathiasMeuleman) [inline resources/options.js]: Please consistantly name your functions. Above you used SubOptions, here it is Subelements (single vs double capital)

File: resources/options.js [hunk 8]
```diff
@@ -133,16 +121,40 @@ function elementGenerator(index, array){
...
+                dynamicalyHideOrShowSubelements(tohide);
+                saveOptions();
+            });
+        })(onDisableHideAndDisable[i]);
+    }
+}
+
+function dynamicalyHideOrShowSubelements(tohide) {
+    if ($("#" + tohide[0]).prop('checked') && !$("#" + tohide[0]).attr('disabled') ) {
+        for (var s in tohide[1]) {
```
Comment (MathiasMeuleman) [inline resources/options.js]: I would like a more descriptive name than 's'

**Comment by MathiasMeuleman** [comment]
Thanks for the changes. Looking good. If someone can confirm their functionality issues are fixed, this can be merged

**Comment by mpsijm** [comment]
I have tried to break the options page again but I didn't have success in letting it fail. :D Will merge! :)

**Pull request merged**